src/client/src/App/Trades/TradesGrid/TradesGrid.tsx

```diff
@@ -1,60 +1,76 @@
+import {
+  createContext,
+  CSSProperties,
+  forwardRef,
+  PropsWithChildren,
+  useContext,
+} from "react"
+import AutoSizer from "react-virtualized-auto-sizer"
+import { FixedSizeList as List, FixedSizeListProps } from "react-window"
 import styled, { css } from "styled-components"
 
 import { QuoteState } from "@/generated/TradingGateway"
-import { CreditTrade, FxTrade, Trade, TradeStatus } from "@/services/trades"
+import { Trade, TradeStatus } from "@/services/trades"
 
 import { useColDef, useColFields, useTrades$ } from "../Context"
 import { useTableTrades } from "../TradesState"
+import { useTableTradeWithIndex } from "../TradesState/tableTrades"
 import { TableHeadCellContainer } from "./TableHeadCell"
+import { convertRemToPixels, getWidthPercentage } from "./utils"
 
-const TableWrapper = styled.div`
+const Table = styled.div`
+  border-bottom: 15px solid ${({ theme }) => theme.core.darkBackground};
   height: calc(100% - 4.75rem);
-  overflow-x: scroll;
-  overflow-y: scroll;
-  contain: content;
-`
-const Table = styled.table`
-  background-color: ${({ theme }) => theme.core.lightBackground};
-  position: relative;
   width: 100%;
-  min-width: 60rem;
-  border-collapse: separate;
-  border-spacing: 0;
   .visually-hidden {
     display: none;
   }
 `
-const TableHead = styled.thead`
+
+const TableHeadRow = styled.div`
+  position: sticky;
+  top: 0;
+  display: flex;
   font-size: 0.675rem;
   text-transform: uppercase;
-`
-const TableHeadRow = styled.tr`
-  vertical-align: center;
+  z-index: 1;
   height: 2rem;
+  background-color: ${({ theme }) => theme.core.lightBackground};
 `
 
 const highlightBackgroundColor = css`
   animation: ${({ theme }) => theme.flash} 1s ease-in-out 3;
 `
 
-const TableBodyRow = styled.tr<{ pending?: boolean; highlight?: boolean }>`
-  &:nth-child(even) {
-    background-color: ${({ theme }) => theme.core.darkBackground};
-  }
+const TableBodyRow = styled.div<{
+  pending?: boolean
```

Comment:
wot's pending for?

Reply:
It's for keeping senior devs reviewing skills sharp